### DISCUSSION TITLE
from torchmetrics.functional import true_negatives, false_positives

### DISCUSSION THREAD

**Discussion by nahondobruno** [Lightning-AI/torchmetrics#1887] [General]
iam new to torchmetrics and Iam inquiring which of these is the right implementation torchmetrics.
Thanks
```python
from torchmetrics.functional import true_negatives, false_positives
TN = true_negatives(mincaps[~no_step_mask], actual_mincaps[~no_step_mask], num_classes=2).sum()
FP = false_positives(mincaps[~no_step_mask], actual_mincaps[~no_step_mask], num_classes=2).sum()
self.mincaps_FP += FP
self.mincaps_TN += TN
```
**OR**

```python
import torchmetrics
TN = torchmetrics.functional.confusion_matrix(mincaps[~no_step_mask], actual_mincaps[~no_step_mask], num_classes=2)[0, 0]
FP = torchmetrics.functional.confusion_matrix(mincaps[~no_step_mask], actual_mincaps[~no_step_mask], num_classes=2)[0, 1]
self.mincaps_FP += FP
self.mincaps_TN += TN
```

**Comment by Borda** [comment]
Welcome to TorchMetrics. If you’re wondering which implementation of metrics like true_negatives or false_positives to use, the recommended way is to import them directly from the `torchmetrics.functional` module as you mentioned:
```py
from torchmetrics.functional import true_negatives, false_positives
```
These functional APIs provide stateless metric calculations that you can use directly on tensors. Alternatively, TorchMetrics also offers metric classes (e.g., `TrueNegatives`, `FalsePositives`) in `torchmetrics` which maintain state across batches and are useful for aggregation during training loops.
If you’re just starting out and want quick metric computations, using the functional APIs is a good approach. As your pipeline grows, consider switching to the metric classes for more integration with PyTorch Lightning or custom training loops.
